sdk/go/pulumi/provider_test.go

```diff
@@ -490,8 +490,14 @@ func TestConstructResult(t *testing.T) {
 	resolvedProps, _, _, err := marshalInputs(state)
 	assert.NoError(t, err)
 
-	assert.Equal(t, resource.NewPropertyMapFromMap(map[string]interface{}{
-		"foo":       "hi",
-		"someValue": "something",
-	}), resolvedProps)
+	assert.Equal(t, resource.PropertyMap{
+		"foo": resource.NewOutputProperty(resource.Output{
```

Comment:
This change requires some explanation...

For those who aren't already familiar, the Go SDK marshaling goes through two layers:

1. First the Go SDK types are marshaled into `resource.PropertyValue` values. That's what this PR is changing.
2. Then when marshaling for gRPC calls, the `resource.PropertyValue` values are marshaled into the gRPC structures. Support for marshaling output values at this layer was already implemented in #7861.

During (2), when the option to `KeepOutputValues` isn't set, the same values as before these changes are returned.

For (1), when marshaling `Output`, the behavior of this PR is currently to always return an output value. (It's up to the `KeepOutputValues` option in (2) to determine whether the output values are kept or not when going across the gRPC boundary.)

This does lead to this unusual case for this test where we have some input values that are `StringOutput` that are being passed to `marshalInputs`, and since we're now returning `Output` as an output value, we need to change the expected values for this test.

I can imagine an alternative behavior for `marshalInputs`, where if the `Output`'s value is known, isn't a secret, and doesn't have any dependencies, we could just return the same value as before, so this test wouldn't have to change. We could similarly do that when the output is a secret and doesn't have any dependencies, just return the same value that would be returned previously. And we could even do it for unknowns when there aren't any dependencies. I don't feel strongly about this either way, as it doesn't really affect anything in practice (aside from this test), but curious what other folks think.